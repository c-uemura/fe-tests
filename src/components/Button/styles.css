.btn {
  border-radius: 8px;
  padding: 8px 16px;
  font-size: 16px;
  font-weight: bold;
  color: #fff;
  background-color: blue;
  cursor: pointer;
}
.disabled {
  background-color: gray;
  color: #efefef;
}
