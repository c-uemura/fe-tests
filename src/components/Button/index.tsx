import './styles.css';
interface Props {
  label: string;
  disabled?: boolean;
  onClick: () => void;
  testID?: string;
}

const Button = ({ label, onClick, disabled, testID }: Props) => (
  <div
    className={disabled ? 'btn disabled' : 'btn'}
    data-testid={testID}
    onClick={() => {
      if (!disabled && onClick) {
        onClick();
      }
    }}
  >
    {label}
  </div>
);

export default Button;
