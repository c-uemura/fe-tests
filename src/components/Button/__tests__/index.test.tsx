import { vi } from 'vitest';
import userEvent from '@testing-library/user-event';
import { screen, render } from '@testing-library/react';
import Button from '@/components/Button';

describe('Button', () => {
  it('render element', async () => {
    const mockFn = vi.fn();
    const label = 'Button Example';

    const { baseElement } = render(
      <Button label={label} onClick={mockFn} testID="btn-test" />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('render disabled Button', async () => {
    const mockFn = vi.fn();
    const label = 'Button Example';

    const { baseElement } = render(
      <Button
        label={label}
        disabled={true}
        onClick={mockFn}
        testID="btn-test"
      />
    );
    expect(baseElement).toMatchSnapshot();

    const btn = await screen.getByTestId('btn-test');
    await userEvent.click(btn);
    expect(mockFn).toHaveBeenCalledTimes(0);
  });

  it('should click button', async () => {
    const mockFn = vi.fn();
    const label = 'Button Example';
    render(<Button label={label} onClick={mockFn} testID="btn-test" />);

    const btn = await screen.getByTestId('btn-test');
    await userEvent.click(btn);
    expect(mockFn).toHaveBeenCalled();
  });
});
