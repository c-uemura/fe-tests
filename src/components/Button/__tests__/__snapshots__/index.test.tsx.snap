// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > render disabled Button 1`] = `
<body>
  <div>
    <div
      class="btn disabled"
      data-testid="btn-test"
    >
      Button Example
    </div>
  </div>
</body>
`;

exports[`Button > render element 1`] = `
<body>
  <div>
    <div
      class="btn"
      data-testid="btn-test"
    >
      Button Example
    </div>
  </div>
</body>
`;
